pkg/provider/types/parameters_test.go

```diff
@@ -0,0 +1,499 @@
+package types
+
+import (
+	"reflect"
+	"testing"
+)
+
+func TestGetKeyVaultName(t *testing.T) {
+	tests := []struct {
```

Comment:
This method does trimming. Should we add another test case with whitespaces and validate for Trim?

Reply:
good call! added tests for trim